src/tests/inductive-proofs-small.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  SelfProof,
+  Field,
+  Experimental,
+  isReady,
+  shutdown,
+  Proof,
+} from '../index.js';
+import { tic, toc } from '../examples/zkapps/tictoc.js';
+
+await isReady;
+
+let MaxProofsVerifiedOne = Experimental.ZkProgram({
+  publicInput: Field,
+
+  methods: {
+    baseCase: {
+      privateInputs: [],
+
+      method(publicInput: Field) {
+        publicInput.assertEquals(Field(0));
+      },
+    },
+
+    mergeOne: {
+      privateInputs: [SelfProof],
+
+      method(publicInput: Field, earlierProof: SelfProof<Field, undefined>) {
+        earlierProof.verify();
+        earlierProof.publicInput.add(1).assertEquals(publicInput);
+      },
+    },
+  },
+});
+
+tic('compiling program');
+await MaxProofsVerifiedOne.compile();
+toc();
+
+await testRecursion(MaxProofsVerifiedOne, 1);
+
+async function testRecursion(
+  Program: typeof MaxProofsVerifiedOne,
```

Comment:
Nit: should this be explicit? I think we want to have to change things if the backend changes, right?